src/components/EditableName.jsx

```diff
@@ -1,26 +1,47 @@
-import React from "react";
+import React, { useEffect, useState } from "react";
 import PropTypes from "prop-types";
 import { Grid, IconButton, TextField } from "@mui/material";
 import { Check, Clear, Edit } from "@mui/icons-material";
 
-class EditableName extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      editing: false,
-      newName: "",
-      error: "",
-    };
-  }
+const EditableName = ({ name, setName, number }) => {
+  const [editing, setEditing] = useState(false);
+  const [newName, setNewName] = useState("");
+  const [error, setError] = useState("");
 
-  componentDidMount = () => {
-    const { name } = this.props;
-    this.setState({ newName: name });
+  useEffect(() => {
+    setNewName(name);
+  }, []);
+
+  const hasError = () => {
+    return error !== "";
+  };
+
+  const startEditing = (event) => {
+    event.preventDefault();
+    setEditing(true);
+    setNewName(name);
+  };
+
+  const handleChange = (event) => {
+    setNewName(event.target.value);
+
+    if (newName === null || newName === undefined || newName === "") {
+      setError("Name cannot be blank.");
+    }
+  };
+
+  const submit = () => {
+    setName(number, newName);
+    setEditing(false);
+  };
+
+  const cancel = () => {
+    setEditing(false);
+    setNewName(name);
+    setError("");
   };
 
-  render = () => {
-    const { name } = this.props;
-    const { editing, newName, error } = this.state;
+  const renderEditableName = () => {
     if (editing) {
       // <Form onSubmit={this.submit} inline>
```

Comment:
cleanup while we're here

Reply:
Removed this comment in [4c7681f](https://github.com/unchained-capital/caravan/pull/342/commits/4c7681fe2badd60264d2566255b8660f7e6c4151)